torchdynamo/variables/functions.py

```diff
@@ -263,6 +282,23 @@ def call_function(
         return result
 
 
+def invoke_and_store_as_constant(tx, fn, name, source, options, args, kwargs):
+    def convert(x):
+        if isinstance(x, variables.TensorVariable):
+            return x.proxy.node.meta["example_value"]
+        return x.as_python_constant()
+
+    args = [convert(x) for x in args]
+    kwargs = {k: convert(x) for k, v in kwargs.items()}
+    res = fn(*args, **kwargs)
+    return tx.output.add_submodule(
```

Comment:
This will only work for tensors I believe.  Add a test of a function that returns `.size(1)`.  Might need some logic similar to VariableBuilder.

Reply:
Correct. I was thinking that we should only support tensors in this API. That feels like it works better with the notions around when people would use this, and keeps the whole thing simpler w/r/t to just baking in a single constant result. Adding support for data structures and scalars can be added later if sorely needed. What do you think?

Reply:
We now support non-tensor types properly.